plugins/out_azure/azure.c

```diff
@@ -95,9 +98,27 @@ static int azure_format(const void *in_buf, size_t in_bytes,
         /* Append the time key */
         msgpack_pack_str(&mp_pck, flb_sds_len(ctx->time_key));
         msgpack_pack_str_body(&mp_pck,
-                              ctx->time_key,
-                              flb_sds_len(ctx->time_key));
-        msgpack_pack_double(&mp_pck, t);
+                            ctx->time_key,
+                            flb_sds_len(ctx->time_key));
+
+        if (ctx->time_generated == FLB_TRUE) {
+            /* Append the time value as ISO 8601 */
+            gmtime_r(&tm.tm.tv_sec, &tms);
+            s = strftime(time_formatted, sizeof(time_formatted) - 1,
+                            FLB_PACK_JSON_DATE_ISO8601_FMT, &tms);
```

Comment:
https://docs.microsoft.com/en-us/azure/azure-monitor/logs/data-collector-api#request-headers
```
" The contents of the message field should follow the ISO 8601 format YYYY-MM-DDThh:mm:ssZ."
```

Do we need last character 'Z' ?
`FLB_PACK_JSON_DATE_ISO8601` is `"%Y-%m-%dT%H:%M:%S"`
The format string doesn't contain last 'Z'.

Reply:
Azure is ingesting this format fine at the moment. I'm running the current Patch for a few months now. But I think the best road to go down with is do comply with the documentation. I will update the Patch.

Reply:
Sorry for the confusion the Z is already appended later after the milliseconds. See https://github.com/tobiaskohlbau/fluent-bit/blob/791f388274e0075c6bb487f4240a2ed343f8948b/plugins/out_azure/azure.c#L112

Reply:
Thank you.